src/rabbit_delayed_message_khepri.erl

```diff
@@ -0,0 +1,135 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%%  Copyright (c) 2007-2020 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+%% NOTE that this module uses os:timestamp/0 but in the future Erlang
+%% will have a new time API.
+%% See:
+%% https://www.erlang.org/documentation/doc-7.0-rc1/erts-7.0/doc/html/erlang.html#now-0
+%% and
+%% https://www.erlang.org/documentation/doc-7.0-rc1/erts-7.0/doc/html/time_correction.html
+
+-module(rabbit_delayed_message_khepri).
+-include_lib("rabbit_common/include/rabbit.hrl").
+-include_lib("kernel/include/logger.hrl").
+-include_lib("rabbit_common/include/logging.hrl").
+
+-export([setup/0,
+         get_store_id/0,
+         put/2,
+         delete/1,
+         get/1,
+         match/1,
+         get_many/1]).
+
+-define(RA_SYSTEM, dmx_coordination).
+-define(RA_CLUSTER_NAME, rabbitmq_dmx_metadata).
+-define(RA_FRIENDLY_NAME, "RabbitMQ Delayed Message Exchange metadata store").
+-define(STORE_ID, ?RA_CLUSTER_NAME).
+
+setup() ->
+    RaServerConfig = #{cluster_name => ?RA_CLUSTER_NAME,
+                       friendly_name => ?RA_FRIENDLY_NAME},
+    ok = ensure_ra_system_started(?RA_SYSTEM),
+    case khepri:start(?RA_SYSTEM, RaServerConfig) of
+        {ok, ?STORE_ID} ->
+            wait_for_leader(),
+            %register_projections() would this be needed for dmx?
```

Comment:
don't we need a projection into an ordered ets table to have efficient lookup on the timestamp comparison?

Reply:
Note that this is a very very very rough PR, so I did not expect to see any comments on the code :)! Hence the draft state. I am more interested in comments on the description!